ldk/node/src/hostClients/networkClient.ts

```diff
@@ -1,8 +1,55 @@
+import structpb from 'google-protobuf/google/protobuf/struct_pb';
+import jspb from 'google-protobuf';
 import BaseClient, { GRPCClientConstructor } from './baseClient';
 import { NetworkClient as NetworkGRPCClient } from '../grpc/network_grpc_pb';
 import { NetworkService, HttpRequest, HttpResponse } from './networkService';
 import messages from '../grpc/network_pb';
 
+/**
+ * @param values - a PB struct ListValue (list of Values)
+ * @internal
+ */
+function parseHeaderValues(values: structpb.ListValue): Array<string> {
+  return values.getValuesList().map((value) => {
+    return value.getStringValue();
+  });
+}
+
+/**
+ * @param headersMap - A map of headers
+ * @internal
+ */
+function parseHeadersMap(
+  headersMap: jspb.Map<string, structpb.ListValue>,
+): Map<string, Array<string>> {
+  const headerEntries: Array<[
+    string,
+    Array<string>,
+  ]> = headersMap.getEntryList().map(([key, value]) => {
```

Comment:
There is actually a bug with this. This and some of the other functions actually turn the ListValues straight into Array<string> but don't reflect that in their typing. Confirmed this breaks with both a GripMock setup and the real thing coming from OliveHelps

Reply:
Addressed